database/mysql.go

```diff
@@ -153,26 +149,51 @@ func CreateTestMySQLDB(t *testing.T) *TestMySQLDB {
 		t.Skip("Docker not enabled")
 	}
 
-	mySQLTestDBSetup.Do(func() {
-		var err error
-		sharedMySQLConfig, err = findOrLaunchMySQLContainer()
-		require.NoError(t, err)
-	})
+	containerName := "mysql-test-db"
+	pool, err := dockertest.NewPool("")
+	require.NoError(t, err)
+
+	mysqlConfig := MySQLConfig{
+		User:     "moov",
+		Password: "secret",
+	}
 
-	dbName, err := CreateTemporaryDatabase(sharedMySQLConfig)
+	resource, err := findOrLaunchMySQLContainer(pool, containerName, mysqlConfig)
 	require.NoError(t, err)
+	mysqlConfig.Address = fmt.Sprintf("tcp(localhost:%s)", resource.GetPort("3306/tcp"))
 
-	dbConfig := &DatabaseConfig{
-		DatabaseName: dbName,
-		MySQL:        sharedMySQLConfig.MySQL,
+	dbName, err := CreateTemporaryDatabase(mysqlConfig)
+	require.NoError(t, err)
+
+	dbURL := fmt.Sprintf("%s:%s@%s/%s",
+		mysqlConfig.User,
+		mysqlConfig.Password,
+		mysqlConfig.Address,
+		dbName,
+	)
+
+	var db *sql.DB
+	err = pool.Retry(func() error {
```

Comment:
Vince, good idea. From one point of view this code makes things simple. And I like it. We rely on docker to handle concurrent calls to Docker API (create container and find container).

But somehow I feel that this code is hiding "concurrency aspect" and may create some false assumptions... for example `err = pool.Retry(func() error {` will be called for each test and at some point when container is ready some tests may wait extra time (5+ seconds) in `Retry` method (they are async and `Retry` has exponential backoff). With `sync.Once` we know that container is ready.

I'm not sure we are adding more value with further changes to base/database. I'll wait for comments from Adam on this and then decide.

Reply:
Good point, I've updated the code to only call `pool.Retry` when the container is first created.

In general, we should try to avoid global variables when we can since they can introduce unexpected behavior and hide dependencies.